src/cmd/action.go

```diff
@@ -0,0 +1,171 @@
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/creasty/defaults"
+	"github.com/opslevel/opslevel-go/v2023"
+
+	"github.com/opslevel/cli/common"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var actionType string
+
+var createActionCmd = &cobra.Command{
+	Use:   "action --type=$ACTION_TYPE",
+	Short: "Create an action",
+	Long:  "Create an action",
+	Example: `
+cat << EOF | opslevel create action --type=webhook -f -
+name: "Page The On Call"
+description: "Pages the On Call"
+webhookUrl: "https://api.pagerduty.com/incidents"
+httpMethod: "POST"
+headers:
+  accept: "application/vnd.pagerduty+json;version=2"
+  authorization: "Token token=XXXXXXXXXXXXX"
+  from: "someone@example.com"
+payload: |
+  {
+      "incident": {
+          "type": "incident",
+          "title": "{{manualInputs.IncidentTitle}}",
+          "service": {
+            "id": "{{ service | tag_value: 'pd_id' }}",
+            "type": "service_reference"
+          },
+          "body": {
+            "type": "incident_body",
+            "details": "Incident triggered from OpsLevel by {{user.name}} with the email {{user.email}}. {{manualInputs.IncidentDescription}}"
+          }
+      }
+  }
+EOF`,
+	Run: func(cmd *cobra.Command, args []string) {
+		switch actionType {
+		case "webhook":
+			input, err := readWebhookActionCreateInput()
+			cobra.CheckErr(err)
+			result, err := getClientGQL().CreateWebhookAction(*input)
+			cobra.CheckErr(err)
+			fmt.Printf("created webhook action: %s\n", result.Id)
+		default:
+			err := errors.New("unknown action type: " + args[0])
+			cobra.CheckErr(err)
+		}
+	},
+}
+
+var getActionCmd = &cobra.Command{
+	Use:        "action ID",
+	Short:      "Get details about an action",
+	Long:       "Get details about an action",
+	Args:       cobra.ExactArgs(1),
+	ArgAliases: []string{"ID"},
+	Run: func(cmd *cobra.Command, args []string) {
+		key := args[0]
+		action, err := getClientGQL().GetCustomAction(*opslevel.NewIdentifier(key))
+		cobra.CheckErr(err)
+		common.PrettyPrint(action)
+	},
+}
+
+var listActionCmd = &cobra.Command{
+	Use:     "action",
+	Aliases: []string{"actions"},
+	Short:   "List actions",
+	Long:    "List actions",
+	Run: func(cmd *cobra.Command, args []string) {
+		resp, err := getClientGQL().ListCustomActions(nil)
+		cobra.CheckErr(err)
+		list := resp.Nodes
+		if isJsonOutput() {
+			common.JsonPrint(json.MarshalIndent(list, "", "    "))
+		} else {
+			w := common.NewTabWriter("ID", "NAME", "HTTP_METHOD", "WEBHOOK_URL")
+			for _, item := range list {
+				fmt.Fprintf(w, "%s\t%s\t%s\t%s\n", item.Id, item.Name, item.HTTPMethod, item.WebhookURL)
+			}
+			w.Flush()
+		}
+	},
+}
+
+var updateActionCmd = &cobra.Command{
+	Use:   "action ID",
```

Comment:
If Get supports `ID or ALIAS` but Update does not - that feels weird.  Is this true?

Maybe we need to paper over this in the CLI - lookup the action by Alias?

Not sure.

Reply:
I noticed something interested with the GraphQL API when looking into this.

The mutation `CustomActionsWebhookActionUpdateInput!` doesn't have an ALIAS as a parameter at all, only ID. Same with `CustomActionsWebhookActionCreateInput!`.

The query for `CustomActionsExternalAction!` does take an `IdentifierInput!` where you can pass an ALIAS or ID. But since you can never create or update an ALIAS into an action, there's no point in using it, since there would never be an Action object where the `Aliases` string array isn't empty. This is the case in Circular.

In that case I think it's best to ignore the `get by alias` and `update by alias` functionality.